storage/kvstore.go

```diff
@@ -277,6 +277,11 @@ func (s *store) Compact(rev int64) (<-chan struct{}, error) {
 
 func (s *store) Hash() (uint32, error) {
 	s.b.ForceCommit()
+	s.mu.Lock()
```

Comment:
i think the test code first calls compact and waits for its finish, then call Hash(). So will this actually make a change?